### ISSUE TITLE
ESP-NOW: frame checksum is not validated (IDFGH-8072)

### ISSUE THREAD

**Issue by ceribus** [espressif/esp-idf#9575]
## Environment

- Chip used: ESP32-C3
- IDF version: v4.4.2

## Problem Description

Frames received by ESP-NOW with invalid FCS (checksum) are not discarded.

### Expected Behavior

Frames with invalid FCS should be discarded.

### Actual Behavior

Frames with invalid FCS are accepted and acknowledged.

![image](https://user-images.githubusercontent.com/20140455/185222337-9e68d45c-4b87-4810-bda7-4805aff15677.png)
![image](https://user-images.githubusercontent.com/20140455/185222088-b3a20a72-db25-4712-b769-3b507866aa40.png)

### Steps to reproduce
Use the firmware and scripts from https://github.com/ceribu/esp-now-header-validation/blob/main/invalid-fcs.py

**Comment by MaxwellAlan** [comment]
Hi @ceribu 

Can you share the type of your wireless cart that you use with scapy to send FCS error packets, when I try your script, the tx packet's FCS part will be modified by the wifi driver.

Thanks.

**Comment by MaxwellAlan** [comment]
> Intel Wireless 8265

Yeah, I try with Kali 21.2 with tplink TL-WN722N and rtl8192cu , both can not tx FCS error pkts. I will try with Inter AX200 at first.
